### DISCUSSION TITLE
useDebounce is wrong

### DISCUSSION THREAD

**Discussion by zampoukos** [primefaces/community#592] [PrimeReact] [answered]
useDebounce should wait for a time lets say 400ms before updating the value **from the last change**, not the first one. 
In this [example ](https://primereact.org/hooks/usedebounce/) if you hold down a key, the debounced value changes every 400ms.
It should only be updated when 400ms pass **if the user stops writing**

**Comment by melloware** [comment, marked as answer]
Issue created: https://github.com/primefaces/primereact/issues/5490

**Reply by melloware** [reply]
Fixed for next release.

**Reply by zampoukos** [reply]
Happy holidays, sorry for bothering you again at a time like this but do we have an approximate date for the next release where this will be fixed?

**Reply by melloware** [reply]
PrimeTek is working hard on Accessibility updates so I am not privvy to their internal timelines for releases. Sorry!
